Assets/HoloToolkit/Utilities/Scripts/BoundingBoxHelper.cs

```diff
@@ -0,0 +1,327 @@
+        {
+            boundsPoints.Clear();
+            for (int i = 0; i < rawBoundingCorners.Count; ++i)
+            {
+                boundsPoints.Add( target.transform.localToWorldMatrix.MultiplyPoint(rawBoundingCorners[i]) );
+            }
+
+            worldBoundingCorners.Clear();
+            worldBoundingCorners.AddRange(boundsPoints);
+        }
+    }
+
+    /// <summary>
+    /// This function gets the untransformed bounding box corner points of a GameObject.
+    /// </summary>
+    /// <param name="target"></param>
+    /// <param name="ignoreLayers"></param>
+    public void GetRawBBCorners(GameObject target, LayerMask ignoreLayers)
+    {
+        targetObject = target;
+
+        GameObject clone = GameObject.Instantiate(targetObject);
+        clone.transform.localRotation = Quaternion.identity;
+        clone.transform.position = Vector3.zero;
+        clone.transform.localScale = new Vector3(1, 1, 1);
+        Renderer[] renderers = clone.GetComponentsInChildren<Renderer>();
+        rawBoundingCorners.Clear();
+        rawBoundingCornersObtained = false;
+        for (int i = 0; i < renderers.Length; ++i)
+        {
+            var rendererObj = renderers[i];
+            if (ignoreLayers == (1 << rendererObj.gameObject.layer | ignoreLayers))
+            {
+                continue;
+            }
+            Vector3[] corners = null;
+            rendererObj.bounds.GetCornerPositionsFromRendererBounds(ref corners);
+            AddAABoundingBoxes(rawBoundingCorners, corners);
+        }
+
+        GameObject.Destroy(clone);
+
+        if (rawBoundingCorners != null && rawBoundingCorners.Count >= 4)
+        {
+            rawBoundingCornersObtained = true;
+        }
+    }
+
+    /// <summary>
+    /// this function gets the indices of the bounding cube corners that make up a face.
+    /// </summary>
+    /// <param name="index">the face index of the bounding cube 0-5</param>
+    /// <returns>an array of four integer indices</returns>
+    public int[] GetFaceIndices(int index)
+    {
+        switch (index)
+        {
+            case 0:
+                return new int[] { 0, 1, 3, 2 };
+            case 1:
+                return new int[] { 1, 5, 7, 3 };
+            case 2:
+                return new int[] { 5, 4, 6, 7 };
+            case 3:
+                return new int[] { 4, 0, 2, 6 };
+            case 4:
+                return new int[] { 6, 2, 3, 7 };
+            case 5:
+                return new int[] { 1, 0, 4, 5};
+        }
+
+        return new int[0];
+    }
+
+    /// <summary>
+    /// This function returns the midpoints of each of the edges of the face of the bounding box
+    /// </summary>
+    /// <param name="index">the index of the face of the bounding cube- 0-5</param>
+    /// <returns>four Vector3 points</returns>
+    public Vector3[] GetFaceEdgeMidpoints(int index)
+    {
+        Vector3[] corners = GetFaceCorners(index);
+        Vector3[] midpoints = new Vector3[4];
```

Comment:
this could also be a readonly field if we're only ever updating the same 4 array values.